app/components/Sidebar/components/StarredLink.tsx

```diff
@@ -40,25 +40,35 @@ function StarredLink({ star }: Props) {
   const { documentId, collectionId } = star;
   const collection = collections.get(collectionId);
   const locationSidebarContext = useLocationState();
-  const sidebarContext = useSidebarContext();
+  const starType = star.documentId ? "document" : "collection";
+  const sidebarContext = starredSidebarContext(
+    starType === "document" ? star.documentId! : star.collectionId
```

Comment:
```suggestion
    star.documentId ?? star.collectionId
```

?

Reply:
Yes, that works. Added the `starType` for readability - will remove it.